### PR TITLE
app_flutter uses CanvasKit

### PR DESCRIPTION
**Pull request by CaseyHillers**
CanvasKit achieves 60 fps for the build dashboard.

`deploy.dart` - Added profile mode flag, use skia
`app_flutter` - `CommitBox` shows author initial instead of image, `StatusGrid` reverted back to `GridView` implementation (reverting https://github.com/flutter/cocoon/pull/518)

## Open Issues
- ~On HTTPS, an unsecured script (font) is loaded. This blocks running the rest of the Flutter app until it is allowed to load.~ - Fixed in https://github.com/flutter/engine/pull/14086

## Preview
Demo: https://testchillers-dot-flutter-dashboard.appspot.com/build.html#/

### CODE REVIEW

**Comment by CaseyHillers** [comment]
Waiting on https://github.com/flutter/engine/pull/14086 to fix this issue

**Comment by CaseyHillers** [comment]
Waiting on https://github.com/flutter/engine/pull/14086 to be rolled into the framework.

**Comment by jonahwilliams** [comment]
Opening up the test page, there seems to be a lot of errors in the console:

```
Another exception was thrown: Instance of 'minified:az<void>'
73main.dart.js:1933 Another exception was thrown: Instance of 'minified:az<void>'
main.dart.js:1933 Another exception was thrown: Instance of 'minified:az<void>'
55main.dart.js:1933 Another exception was thrown: Instance of 'minified:az<void>'
main.dart.js:1933 Another exception was thrown: Instance of 'minified:az<void>'
13main.dart.js:1933 Another exception was thrown: Instance of 'minified:az<void>'
```

**Comment by digiter** [comment]
How much fps is the old one?

**Comment by jonahwilliams** [comment]
The gridview was about ~5 fps on my desktop machine. Switching to the listview of listviews bumped it up to the low teens, but that is a bit of a hack that brought its own problems

**Comment by CaseyHillers** [comment]
@digiter for some reason the CanvasKit just shows consistently at 1-3 FPS in the Chrome DevTools, but it is running at a much higher FPS.

From https://github.com/flutter/cocoon/pull/485, DevTools put it at an average of ~40 FPS. However, the performance dropped down to 1 FPS on moving.

**Comment by CaseyHillers** [comment]
@jonahwilliams I noticed this being thrown whenever the mouse moves. I am going to see if switching back to `GridView` fixes it.

**Comment by jonahwilliams** [comment]
That is probably due to https://github.com/flutter/flutter/issues/45510

**Comment by CaseyHillers** [comment]
@jonahwilliams I think this is a separate issue. https://github.com/flutter/flutter/issues/45510 is reproducible on https://flutter-dashboard.appspot.com/build.html#/ but does not throw this exception. This should be a new issue. I am narrowing down the issue and the cause to file it.

Changing back to the GridView implementation still had the issue (although the code was much cleaner :)

**Comment by CaseyHillers** [comment]
Follow up: It's a race condition, but infrequently works as expected.

**Comment by CaseyHillers** [comment]
I filed the issue https://github.com/flutter/flutter/issues/46027 since it was different than the one previously mentioned that's a release blocker.

File: app_flutter/lib/commit_box.dart [hunk 1]
```diff
@@ -33,15 +33,32 @@ class _CommitBoxState extends State<CommitBox> {
...
   Widget build(BuildContext context) {
+    int authorHash = widget.commit.author.hashCode;
     return SizedBox(
       width: StatusGrid.cellSize,
       height: StatusGrid.cellSize,
       child: GestureDetector(
         onTap: _handleTap,
+        // TODO(chillers): Show a Network Image. https://github.com/flutter/flutter/issues/45955
+        // CanvasKit currently cannot render a NetworkImage because of CORS issues.
+        // As a work around, we just show the first letter of the contributor's username.
```
Comment (digiter) [inline app_flutter/lib/commit_box.dart]: Unrelated to code: could you do a PSA in the infra discord channel to explain this change when deployed?

Comment (CaseyHillers) [reply app_flutter/lib/commit_box.dart]: SGTM

**Comment by CaseyHillers** [comment]
Since this PR is going be staged for a bit, I added reverting the `StatusGrid` back to the `GridView` implementation which is much simpler.

**Comment by CaseyHillers** [comment]
@digiter I added back the `GridView` implementation of the `StatusGrid`. Would you mind taking a look to see if it looks good?

**Comment by yjbanov** [comment]
I'd be curious to see a performance trace from a profile build. Would it be possible to have a profile build hosted side-by-side with the release build?

**Comment by CaseyHillers** [comment]
@yjbanov of course! I am deploying both now that this PR is approved.

**Comment by yjbanov** [comment]
@CaseyHillers Awesome! What URL should I hit to reach the profile build?

**Comment by CaseyHillers** [comment]
@yjbanov 

http://profiledom.flutter-dashboard.appspot.com/build.html#/ - Default
  - Note: Uses a ListView of ListViews and syncs the ScrollControllers

http://profileskia.flutter-dashboard.appspot.com/build.html#/ - CanvasKit
  - Note: Uses a GridView

**Comment by yjbanov** [comment]
Thanks! I found a couple of things we can do to improve frame rate in the canvaskit backend. Working on a PR.
